metaflow/plugins/kfp/tests/sample_flows/complex_flow_1.py

```diff
@@ -0,0 +1,55 @@
+from metaflow import FlowSpec, step
+
+
+class ComplexFlow(FlowSpec):
+    @step
+    def start(self):
+        self.next(self.br1, self.br2)
+
+    @step
+    def br1(self):
+        self.next(self.br11, self.br12)
+
+    @step
+    def br2(self):
+        self.next(self.br21, self.br22)
+
+    @step
+    def br11(self):
+        self.next(self.join1)
+
+    @step
+    def br12(self):
+        self.next(self.join1)
+
+    @step
+    def br21(self):
+        self.next(self.join2)
+
+    @step
+    def br22(self):
+        self.next(self.join2)
+
+    @step
+    def join1(self, br1_inp):
+        self.var1 = 200
+        self.next(self.join3)
```

Comment:
Let's test setting state in a node before the split, after the split and in the join, and validate the state is correct in each.